src/main/java/org/gradle/profiler/studio/StudioGradleClient.java

```diff
@@ -98,26 +96,16 @@ public StudioGradleClient(GradleBuildConfiguration buildConfiguration, Invocatio
 
     @Override
     public void close() {
-        try {
+        try (Server studioPluginServer = this.studioPluginServer;
+             Server studioAgentServer = this.studioAgentServer;
+             Closeable uninstallPlugin = studioPluginInstaller::uninstallPlugin) {
```

Comment:
This is a lot better, though I feel this cast to `Closable` is still weird. It's at least a bit less magical than when you passed it directly as a parameter. Anyway, I think it's good now.